samples/bluetooth/central_and_peripheral_hr/Kconfig

```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+source "Kconfig.zephyr"
+
+config SETTINGS
+	default y
+
+config ZMS
+	default y if SOC_FLASH_NRF_RRAM
```

Comment:
Do we really need to add another config file to every sample? If we always want to use the new storage system for settings if we have RRAM can we not set these defaults in the ZMS/NVS subsystems?

Reply:
ZMS or NVS subsystems should be board agnostic, it works on all flash technologies (RRAM, MRAM, classical flash, ...)

Reply:
Can we define these defaults at the board level?

Reply:
I have tried to add this but zephyr does not have good mechanism to do it.
You can enable FS on the board soc/level but if you don't use settings then this FS will likely be a dead code.
You cannot enable it based on settings being enabled as settings backend is set based on fs being used. You have a circular dependency.

I would recommend to change interface to add information about fs in dts.

Reply:
You can probably make Setting backends (`choice SETTINGS_BACKEND`) select their dependency (their target FS - e.g. `select ZMS`). Now, these choice options use `depends on` syntax so this requires modification of  `zephyr/subsys/settings/Kconfig`. Then, you would pick the recommended (default) backend in the Kconfig.default file in the SoC directory (e.g. `zephyr/soc/nordic/nrf54l/Kconfig.defconfig`).

However, this method has its implications and changes the way Settings subsystem is configured. Right now, the FS selection cause the Settings backend default to change

Reply:
I liked the DTS approach initially that was suggested by Jamie. However, then we realized that relying on the DTS partition node wouldn't probably work for targets that use Partition Manager (nRF52 Series / nRF53 Series / nRF54L Series)

Reply:
It will work for the default, assuming that is the same as what PM is using e.g. if you use nrf54l15dk and storage is internal RRAM in DTS and PM is the same, it will work, it will not work if storage is internal RRAM in DTS but PM is using some external QSPI flash device